src/mds/MDLog.cc

```diff
@@ -626,7 +626,12 @@ void MDLog::trim(int m)
     op_prio = CEPH_MSG_PRIO_HIGH;
 
   unsigned new_expiring_segments = 0;
-
+  
+  unsigned expiring_segments_size = g_conf()->mds_log_max_expiring_segments;
+  if(segments.size() > pre_segments_size && expiring_segments_size > 0)
+    expiring_segments_size = std::max(expiring_segments_size,static_cast<unsigned>(segments.size() - pre_segments_size));
+  pre_segments_size = segments.size();
```

Comment:
pre_segments_size is not used

Reply:
It's used here.
expiring_segments_size = std::max(expiring_segments_size,static_cast<unsigned>

Reply:
It's used here.
expiring_segments_size = std::max(expiring_segments_size,static_cast

Reply:
only set pre_segments_size when replay is done. decrease it (until it reach 0) when segments expired.

Reply:
ok